endpoints/vault/earnings/index.ts

```
@@ -0,0 +1,160 @@
+import fetch from "node-fetch";
+import BigNumber from "bignumber.js";
+import wrap from "../../../utils/wrap";
+import { SubGraphUrl } from "../../../settings/env";
+
+interface RepsonseContainer {
+  data: AccountContainer;
+}
+
+interface AccountContainer {
+  account: AccountEvents;
+}
+
+interface AccountEvents {
+  deposits: VaultEvent[];
+  tokensReceived: VaultEvent[];
+  tokensSent: VaultEvent[];
+  withdrawals: VaultEvent[];
+  vaultPositions: VaultEvent[];
+}
+
+interface VaultEvent {
+  tokenAmount: string;
+  vault: VaultIdContainer;
+}
+
+interface VaultIdContainer {
+  id: string;
+}
+
+interface Event {
+  pathParameters: PathParameters;
+}
+
+interface PathParameters {
+  userAddress: string;
+}
+
+const getAccountEarnings = async (address: string) => {
+  const query = `
+  {
+    account(id: "${address}") {
+      tokensSent: sharesSent {
+        tokenAmount
+        vault {
+          id
+        }
+      }
+      tokensReceived: sharesReceived {
+        tokenAmount
+        vault {
+          id
+        }
+      }
+      deposits {
+        tokenAmount
+        vault {
+          id
+        }
+      }
+      withdrawals {
+        tokenAmount
+        vault {
+          id
+        }
+      }
+      vaultPositions {
+        tokenAmount: balanceTokens
+        vault {
+          id
+        }
+      }    
+    }
+  }
+  `;
+
+  const response = await fetch(SubGraphUrl || "", {
```

would prefer to throw a 500 error if the `SubGraphUrl` is empty but not sure how to do so